### PR TITLE
feat!: make `error` and `loading` properties readonly on `usePaymentProvider` composable

### PR DESCRIPTION
**Pull request by VitorLuizC**
## Description
- **[BREAKING CHANGE]** Make the `error` and `loading` properties readonly (they're refs);
- **[BREAKING CHANGE]** Rename `SetPaymentMethodOnCartInputs` interface to `PaymentMethodParams`;
- Add types for `usePaymentProvider`;
- Write TSDocs for `getAvailablePaymentMethods` API method and its GraphQL query;
- Write TSDocs for `setPaymentMethodOnCart` API method and its GraphQL query;
- Write TSDocs for `usePaymentProvider` composable and its types;
- Convert `usePaymentProvider`, `getAvailablePaymentMethods`, `setPaymentMethodOnCart` and other root-level functions into function declarations (since it's better for the docs generator);
- Export `usePaymentProvider` types in `~/composables`;
- Fix the `usePaymentProvider` commands that were using `context` instead of `context.app` and return a list instead of a single payment method.

## Related Issue
https://vsf.atlassian.net/browse/M2-386

## Motivation and Context
Improves the DX of the `usePaymentProvider` composable.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by sethidden — APPROVED** [review]

File: packages/api-client/src/api/getAvailablePaymentMethods/index.ts [hunk 1]
```diff
@@ -2,31 +2,32 @@ import { ApolloQueryResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
 import { Context } from '../../types/context';
 import GuestAvailablePaymentMethods from './GuestAvailablePaymentMethods';
-import {
-  GuestAvailablePaymentMethodsQuery,
-  GuestAvailablePaymentMethodsQueryVariables,
-} from '../../types/GraphQL';
+import { GuestAvailablePaymentMethodsQuery, GuestAvailablePaymentMethodsQueryVariables } from '../../types/GraphQL';
```
Comment (bartoszherba) [inline packages/api-client/src/api/getAvailablePaymentMethods/index.ts:5]:
```suggestion
import type { GuestAvailablePaymentMethodsQuery, GuestAvailablePaymentMethodsQueryVariables } from '../../types/GraphQL';
```

File: packages/api-client/src/api/getAvailablePaymentMethods/index.ts [hunk 2]
```diff
@@ -2,31 +2,32 @@ import { ApolloQueryResult } from '@apollo/client/core';
...
   };
 
-  const { paymentMethods } = context.extendQuery(
-    customQuery,
-    {
-      paymentMethods: {
-        query: GuestAvailablePaymentMethods,
-        variables: defaultVariables,
-      },
+  const { paymentMethods } = context.extendQuery(customQuery, {
```
Comment (bartoszherba) [inline packages/api-client/src/api/getAvailablePaymentMethods/index.ts:25]: What rules do you use to format code?

File: packages/api-client/src/api/setPaymentMethodOnCart/index.ts [hunk 3]
```diff
@@ -1,34 +1,34 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import setPaymentMethodOnCart from './setPaymentMethodOnCart';
-import {
-  SetPaymentMethodOnCartInput,
-  SetPaymentMethodOnCartMutation,
-  SetPaymentMethodOnCartMutationVariables,
-} from '../../types/GraphQL';
+import setPaymentMethodOnCartMutation from './setPaymentMethodOnCart';
+import { SetPaymentMethodOnCartInput, SetPaymentMethodOnCartMutation, SetPaymentMethodOnCartMutationVariables } from '../../types/GraphQL';
```
Comment (bartoszherba) [inline packages/api-client/src/api/setPaymentMethodOnCart/index.ts:4]:
```suggestion
import type { SetPaymentMethodOnCartInput, SetPaymentMethodOnCartMutation, SetPaymentMethodOnCartMutationVariables } from '../../types/GraphQL';
```

File: packages/theme/composables/usePaymentProvider/commands/setPaymentMethodOnCartCommand.ts [hunk 4]
```diff
@@ -1,17 +1,13 @@
...
-      .api
-      .setPaymentMethodOnCart(paymentMethodParams);
+  execute: async (context: Context, params: PaymentMethodParams): Promise<AvailablePaymentMethod[]> => {
+    const { data } = await context.app.$vsf.$magento.api.setPaymentMethodOnCart(params);
 
-    return data
-      .setPaymentMethodOnCart
-      .cart
-      .available_payment_methods;
+    return data.setPaymentMethodOnCart.cart.available_payment_methods ?? [];
```
Comment (bartoszherba) [inline packages/theme/composables/usePaymentProvider/commands/setPaymentMethodOnCartCommand.ts:11]:
```suggestion
    return data?.setPaymentMethodOnCart?.cart?.available_payment_methods ?? [];
```

**Review by bartoszherba — CHANGES_REQUESTED** [review]

File: packages/api-client/src/api/getAvailablePaymentMethods/index.ts (hunk 2 above)
Comment (VitorLuizC) [reply packages/api-client/src/api/getAvailablePaymentMethods/index.ts:25]: Just executed Prettier and ESLint autofix in sequence. I'm using VSCode extensions for both, so these are my Ctrl + F.

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**